extension/json/json_functions/json_transform.cpp

```diff
@@ -689,6 +689,55 @@ bool TransformToJSON(yyjson_val *vals[], yyjson_alc *alc, Vector &result, const
 	return true;
 }
 
+bool TransformValueIntoUnion(yyjson_val **vals, yyjson_alc *alc, Vector &result, const idx_t count,
+                             JSONTransformOptions &options) {
+	auto type = result.GetType();
+
+	auto fields = UnionType::CopyMemberTypes(type);
+	vector<string> names;
+	for (const auto &field : fields) {
+		names.push_back(field.first);
+	}
+
+	for (idx_t i = 0; i < count; i++) {
+		if (!yyjson_is_obj(vals[i])) {
+			if (options.error_missing_key) {
+				throw InvalidInputException("Expected an object representing a union");
+			} else {
+				result.SetValue(i, Value(nullptr));
+				continue;
+			}
+		}
+
+		yyjson_val *key, *val;
+		yyjson_obj_iter iter;
+		yyjson_obj_iter_init(vals[i], &iter);
+		key = yyjson_obj_iter_next(&iter);
+		if (key == nullptr) {
```

Comment:
`key` should not be able to ever be `nullptr` when using `yyjson_obj_foreach`. Perhaps it's worth checking `yyjson_get_len` here to check if it's an empty object before going into the loop